internal/cmd/egctl/config.go

```diff
@@ -0,0 +1,159 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package egctl
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/types"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"sigs.k8s.io/yaml"
+
+	"github.com/envoyproxy/gateway/internal/cmd/options"
+	kube "github.com/envoyproxy/gateway/internal/kubernetes"
+)
+
+var (
+	output       string
+	podName      string
+	podNamespace string
+)
+
+const (
+	adminPort     = 19000   // TODO: make this configurable until EG support
+	containerName = "envoy" // TODO: make this configurable until EG support
+)
+
+func NewConfigCommand() *cobra.Command {
+	cfgCommand := &cobra.Command{
+		Use:     "config",
+		Aliases: []string{"c"},
+		Short:   "Retrieve proxy configuration.",
+		Long:    "Retrieve information about proxy configuration from envoy proxy and gateway.",
+	}
+
+	cfgCommand.AddCommand(proxyCommand())
+
+	flags := cfgCommand.Flags()
+	options.AddKubeConfigFlags(flags)
+
+	cfgCommand.PersistentFlags().StringVarP(&output, "output", "o", "json", "One of 'yaml' or 'json'")
+	cfgCommand.PersistentFlags().StringVarP(&podNamespace, "namespace", "n", "envoy-gateway", "Namespace where envoy proxy pod are installed.")
+
+	return cfgCommand
+}
+
+func proxyCommand() *cobra.Command {
+	c := &cobra.Command{
+		Use:     "envoy-proxy",
+		Aliases: []string{"proxy"},
+		Long:    "Retrieve information from envoy proxy.",
+	}
+
+	c.AddCommand(allConfigCmd())
+
+	return c
+}
+
+func allConfigCmd() *cobra.Command {
+
+	allConfigCmd := &cobra.Command{
```

Comment:
thoughts on adding `-A` to get information from all pods in all namespaces ?

Reply:
I never thought of such a user case, can you share more details?
it would be better to track that with a ticket.

Reply:
I think it makes sense, `-A` allows users to get info from all pods in all namespaces, like `kubectl get <resources> -A`.

Reply:
yah helps users like me who dont want to run an extra `kubectl get po -A` cmd :)  to get pod name